tests/shared/permissions.spec.js

```diff
@@ -32,70 +34,32 @@ describe('Shared: permissions', function () {
         this.osPlatformStub.restore()
       })
 
-      it('should return an empty array if no environment', function () {
-        m.chai.expect(permissions.getEnvironmentCommandPrefix()).to.deep.equal([])
-      })
-
-      it('should return an empty array if the environment is an empty object', function () {
-        m.chai.expect(permissions.getEnvironmentCommandPrefix({})).to.deep.equal([])
-      })
-
-      it('should create an environment command prefix out of one variable', function () {
-        m.chai.expect(permissions.getEnvironmentCommandPrefix({
-          FOO: 'bar'
-        })).to.deep.equal([
-          'set',
-          'FOO=bar&&',
-          'call'
-        ])
-      })
-
-      it('should create an environment command prefix out of many variables', function () {
-        m.chai.expect(permissions.getEnvironmentCommandPrefix({
-          FOO: 'bar',
-          BAR: 'baz',
-          BAZ: 'qux'
-        })).to.deep.equal([
-          'set',
-          'FOO=bar&&',
-          'set',
-          'BAR=baz&&',
-          'set',
-          'BAZ=qux&&',
-          'call'
-        ])
-      })
-
-      it('should ignore undefined and null variable values', function () {
-        m.chai.expect(permissions.getEnvironmentCommandPrefix({
-          FOO: null,
-          BAR: 'qux',
-          BAZ: undefined
-        })).to.deep.equal([
-          'set',
-          'BAR=qux&&',
-          'call'
-        ])
-      })
-
-      it('should stringify number values', function () {
-        m.chai.expect(permissions.getEnvironmentCommandPrefix({
-          FOO: 1,
-          BAR: 0,
-          BAZ: -1
-        })).to.deep.equal([
-          'set',
-          'FOO=1&&',
-          'set',
-          'BAR=0&&',
-          'set',
-          'BAZ=-1&&',
-          'call'
-        ])
+      it('should escape environment variables and arguments', function () {
+        m.chai.expect(
+          permissions.createLaunchScript(
+            "C:\\Users\\Alice & Bob's Laptop\\\"what\"\\balenaEtcher",
+            [
+              '"a Laser"',
+              'arg1',
+              "'&/ ^ \\",
+              '" $ % *'
+            ],
+            {
+              key: 'value',
+              key2: ' " \' ^ & = + $ % / \\',
+              key3: 8
+            }
+          )
+        ).to.equal(
+          `set "key=value"${os.EOL}` +
+          `set "key2= " ' ^^ ^& = + $ % / \\"${os.EOL}` +
+          `set "key3=8"${os.EOL}` +
+          `"C:\\Users\\Alice & Bob's Laptop\\\\"what\\"\\balenaEtcher" "\\"a Laser\\"" "arg1" "'&/ ^ \\" "\\" $ % *"`
+        )
       })
     })
 
-    describe('given linux', function () {
+    describe('given unix', function () {
```

Comment:
You've changed "linux" -> "unix" here, but a couple of lines below it still says `this.osPlatformStub.returns('linux')` ?
I've no idea whether that's important or not though :wink:

Reply:
osx and linux should behave the same, updated the tests to test both 'linux' and 'darwin'